File: python/netlink/route/links/dummy.py
#
# Copyright (c) 2011 Thomas Graf <tgraf@suug.ch>
#

"""Dummy

"""
from __future__ import absolute_import

__version__ = "1.0"
__all__ = ['init']


from ... import core as netlink
from ..  import capi as capi
class DummyLink(object):
	def __init__(self, link):
        	self._rtnl_link = link

	def brief(self):
        	return 'dummy'
	
def init(link):
	link.dummy = DummyLink(link._rtnl_link)
        return link.dummy

File: lib/route/link/dummy.c
/* SPDX-License-Identifier: LGPL-2.1-only */
/*
 * Copyright (c) 2011 Thomas Graf <tgraf@suug.ch>
 */

/**
 * @ingroup link
 * @defgroup dummy Dummy
 *
 * @details
 * \b Link Type Name: "dummy"
 *
 * @{
 */

#include <netlink-private/netlink.h>
#include <netlink/netlink.h>
#include <netlink-private/route/link/api.h>

static struct rtnl_link_info_ops dummy_info_ops = {
	.io_name		= "dummy",
};

static void __init dummy_init(void)
{
	rtnl_link_register_info(&dummy_info_ops);
}

static void __exit dummy_exit(void)
{
	rtnl_link_unregister_info(&dummy_info_ops);
}

/** @} */
